File: src/main/java/name/hennr/series/stalker/series/SeriesModel.java
package name.hennr.series.stalker.series;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class SeriesModel implements Comparable<SeriesModel> {
    public String imageUrl;

    public SeriesModel() { }

    public SeriesModel(final String imageUrl, final LocalDate nextAirDate) {
        this.imageUrl = imageUrl;
        this.nextAirDate = nextAirDate;
    }

    public LocalDate nextAirDate;

    public String getNextAirDate() {
        if (nextAirDate.isBefore(LocalDate.now())) {
            return "last: " + nextAirDate.format(DateTimeFormatter.ofPattern("dd.MM.yy"));
        } else {
            return "next: " + nextAirDate.format(DateTimeFormatter.ofPattern("dd.MM.yy"));
        }
    }

    @Override
    public int compareTo(SeriesModel seriesModel) {

        if (seriesModel.nextAirDate.isBefore(LocalDate.now()) && this.nextAirDate.isBefore(LocalDate.now())) {
            return seriesModel.nextAirDate.compareTo(nextAirDate);
        } else if (seriesModel.nextAirDate.isAfter(LocalDate.now()) && this.nextAirDate.isAfter(LocalDate.now())) {
            return -1 * seriesModel.nextAirDate.compareTo(nextAirDate);
        } else {
            return seriesModel.nextAirDate.compareTo(this.nextAirDate);
        }
    }
}

File: src/main/java/name/hennr/series/stalker/series/SeriesModel.kt
package name.hennr.series.stalker.series

import java.time.LocalDate
import java.time.format.DateTimeFormatter

class SeriesModel : Comparable<SeriesModel> {
    var imageUrl: String? = null

    constructor() {}
    constructor(imageUrl: String?, nextAirDate: LocalDate?) {
        this.imageUrl = imageUrl
        this.nextAirDate = nextAirDate
    }

    @JvmField
    var nextAirDate: LocalDate? = null
    fun getNextAirDate(): String {
        return if (nextAirDate!!.isBefore(LocalDate.now())) {
            "last: " + nextAirDate!!.format(DateTimeFormatter.ofPattern("dd.MM.yy"))
        } else {
            "next: " + nextAirDate!!.format(DateTimeFormatter.ofPattern("dd.MM.yy"))
        }
    }

    override fun compareTo(seriesModel: SeriesModel): Int {
        return if (seriesModel.nextAirDate!!.isBefore(LocalDate.now()) && nextAirDate!!.isBefore(LocalDate.now())) {
            seriesModel.nextAirDate!!.compareTo(nextAirDate)
        } else if (seriesModel.nextAirDate!!.isAfter(LocalDate.now()) && nextAirDate!!.isAfter(LocalDate.now())) {
            -1 * seriesModel.nextAirDate!!.compareTo(nextAirDate)
        } else {
            seriesModel.nextAirDate!!.compareTo(nextAirDate)
        }
    }
}
